neuralprophet/time_net.py

```diff
@@ -987,6 +988,73 @@ def denormalize(self, ts):
     def train_dataloader(self):
         return self.train_loader
 
+    def get_regressor_coefficients(self):
+        """
+        Retrieves the coefficients for future regressors, lagged regressors, and events.
+
+        Note: The average weight calculation is performed to get a single representative
+        value of the coefficient for a given regressor when there are multiple forecasts
+        or hidden layers.
+
+        Returns
+        -------
+            pd.DataFrame: A DataFrame containing the following columns:
+                - regressor: Name of the regressor.
+                - regressor_mode: Mode of the regressor ('additive' or 'multiplicative').
+                - coef: Average coefficient value for the regressor.
+        """
+        coefficients = []
+
+        # Helper function to calculate the average weight
+        def calculate_average_weight(weight):
```

Comment:
This function applied to both future-reg/events and lagged-reg/AR will be hard to implement correctly. I suggest splitting it into two - one function for model components where each named input/regressor is a single value (future-reg/events) and one for components where an input/regressor consists of multiple values (lagged-reg/AR). The first should return a single coefficient per regressor, while the latter returns multiple coefficients (one per number of lags inputed).

Reply:
Note: to achieve the described outcomes, you may also need to adjust the code below, not just `calculate_average_weight`

Reply:
Yes makes sense! I separated it into two functions and also now return the corresponding lag for each coefficient to improve the usability.